backend/coins/btc/electrum/client/client.go

```diff
@@ -128,58 +131,42 @@ func (client *ElectrumClient) RegisterOnConnectionStatusChangedEvent(onConnectio
 	})
 }
 
-// ServerVersion is returned by ServerVersion().
-type ServerVersion struct {
-	Version         string
-	ProtocolVersion *semver.SemVer
+// electrumxVersion is returned by serverVersion().
+type electrumxVersion struct {
+	software string
+	protocol *semver.SemVer
 }
 
-func (version *ServerVersion) String() string {
-	return fmt.Sprintf("%s;%s", version.Version, version.ProtocolVersion)
+func (v electrumxVersion) String() string {
+	return fmt.Sprintf("%s;%s", v.software, v.protocol)
 }
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
-func (version *ServerVersion) UnmarshalJSON(b []byte) error {
-	slice := []string{}
-	if err := json.Unmarshal(b, &slice); err != nil {
-		return errp.WithContext(errp.Wrap(err, "Failed to unmarshal JSON"), errp.Context{"raw": string(b)})
+// serverVersion makes the "server.version" RPC call.
+// ElectrumX will reply with a success only once. Subsequent calls return
+// a "server.version already sent" error. Electrs doesn't enforce "only once"
+// condition.
+func (client *ElectrumClient) serverVersion() (electrumxVersion, error) {
+	var resp [2]string // [software version, protocol version]
+	err := client.rpc.MethodSync(&resp, "server.version", SoftwareVersion, supportedProtocolVersion)
+	if err != nil {
+		return electrumxVersion{}, err
 	}
-	if len(slice) != 2 {
-		return errp.WithContext(errp.New("Unexpected reply"), errp.Context{"raw": string(b)})
+	semv, err := semver.NewSemVerFromString(resp[1])
+	if err != nil {
+		semv, err = semver.NewSemVerFromString(resp[1] + ".0")
 	}
-	version.Version = slice[0]
-	protocolVersion := slice[1]
-	// We expect the protocolVersion to be either major.minor or major.minor.patch.
-	protocolSemVer, err := semver.NewSemVerFromString(protocolVersion)
 	if err != nil {
-		protocolSemVer, err = semver.NewSemVerFromString(protocolVersion + ".0")
-		if err != nil {
-			return err
-		}
+		semv = &semver.SemVer{} // don't really care; nice to have
```

Comment:
why nice to have? if so, we could drop the response?

should we check the server version and drop the connection if it is newer than 1.4?